docs/getting-started.md

```diff
@@ -1,372 +1,273 @@
-- `build/cargo_target/${toolchain}/debug/jailer`.
+# Run firecracker
+./firecracker/build/cargo_target/${ARCH}-unknown-linux-musl/debug/firecracker \
+    --api-sock "${API_SOCKET}"
+```
 
-If you would like to test a new feature and work with
-dependencies on libraries located in private git repos, you
-can use the `--ssh-keys` flag to specify the paths to your
-public and private SSH keys on the host. Both of them are
-required for git authentication when fetching the repositories.
+In a new terminal (do not close the 1st one):
 
 ```bash
-tools/devtool build --ssh-keys ~/.ssh/id_rsa.pub ~/.ssh/id_rsa
+TAP_DEV="tap0"
+TAP_IP="172.16.0.1"
+MASK_SHORT="/30"
+
+# Setup network interface
+sudo ip link del "$TAP_DEV" 2> /dev/null || true
+sudo ip tuntap add dev "$TAP_DEV" mode tap
+sudo ip addr add "${TAP_IP}${MASK_SHORT}" dev "$TAP_DEV"
+sudo ip link set dev "$TAP_DEV" up
+
+# Setup ip forwarding
+sudo sh -c "echo 1 > /proc/sys/net/ipv4/ip_forward"
+sudo iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE || true
+sudo iptables -D FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT \
+    || true
+sudo iptables -D FORWARD -i tap0 -o eth0 -j ACCEPT || true
+
+# Enable microVM internert access
+sudo iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+sudo iptables -I FORWARD 1 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
+sudo iptables -I FORWARD 1 -i tap0 -o eth0 -j ACCEPT
+
+API_SOCKET="./firecracker.socket"
+LOGFILE="./firecracker.log"
+
+# Create log file
+touch $LOGFILE
+
+# Set log file
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"log_path\": \"${LOGFILE}\",
+        \"level\": \"Debug\",
+        \"show_level\": true,
+        \"show_log_origin\": true
+    }" \
+    "http://localhost/logger"
+
+# Set machine config
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"vcpu_count\": 2,
+        \"mem_size_mib\": 128,
+        \"cpu_template\": \"C3\"
+    }" \
+    "http://localhost/machine-config"
+
+KERNEL="./vmlinux.bin"
+KERNEL_BOOT_ARGS="console=ttyS0 reboot=k panic=1 ipv6.disable=1 i8042.nokbd
+ 8250.nr_uarts=1 random.trust_cpu=on"
+
+# Set boot source
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"kernel_image_path\": \"${KERNEL}\",
+        \"boot_args\": \"${KERNEL_BOOT_ARGS}\"
+    }" \
+    "http://localhost/boot-source"
+
+ROOTFS="./ubuntu-18.04.ext4"
+
+# Set rootfs
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"drive_id\": \"rootfs\",
+        \"path_on_host\": \"${ROOTFS}\",
+        \"is_root_device\": true,
+        \"is_read_only\": false
+    }" \
+    "http://localhost/drives/rootfs"
+
+FC_MAC="06:00:AC:10:00:02"
+
+# Set network interface
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"iface_id\": \"net1\",
+        \"guest_mac\": \"$FC_MAC\",
+        \"host_dev_name\": \"$TAP_DEV\"
+    }" \
+    "http://localhost/network-interfaces/net1"
+
+sleep 0.015s
```

Comment:
mh, instead of sleeping, can we give a bit more context here? Something along the lines of "Wait until you see in your first terminal that the microvm has started. Once it has done so, you can SSH into it using the key downloaded above, e.g [...]`?

Reply:
Added comments on the `sleep`s:

> # API requests are handled asynchronously, it is important the configuration is
> # set, before `InstanceStart`.

and

> # API requests are handled asynchronously, it is important the microVM has been
> # started before we attempt to SSH into it.

respectively.